erpnext/stock/get_item_details.py

```diff
@@ -74,9 +74,17 @@ def get_item_details(args, doc=None, for_validate=False, overwrite_warehouse=Tru
 
 	update_party_blanket_order(args, out)
 
-	
+	# we do not send customer args for price list rate in case of purchase order but we save it for latter use
+	if doc and doc.get('doctype') == 'Purchase Order':
```

Comment:
This is a generic query and hardcoding doctype here doesn't make much sense. We can pass two more args as `is_buying` and `is_selling` based on the doctype it is called from and ignore parties based on the `is_buying` and `is_selling` flag and use the same in item price query as well. This sounds like a more generic approach to me

Reply:
@deepeshgarg007 ok, why noit, but there is test on docytype just few lines before (line 60 for exemple). I apply the same method that are existing into existing code.
But any way I try your way.

Reply:
Update PR is on her way